### DISCUSSION TITLE
RandomZoom on non-zero areas

### DISCUSSION THREAD

**Discussion by maxencewynen** [Project-MONAI/MONAI#4787] [Q&A]
Hi!

I am developing an object detection network in the MONAI framework and liking it a lot so far.

I would like to use a random zoom augmentation (Random Crop + Keep original size by interpolating the result) but make sure the cropped ROI includes at least one of the objects the network is supposed to detect.

Is there an easy way to do this? I could not find any argument in the Zoom transforms, and the Crop transforms do not seem to integrate the keeping the original size and interpolation options either.

Thanks for your help!

**Comment by KumoLiu** [comment]
Hi @maxencewynen,
I think maybe you can use `RandCropByPosNegLabel` with pos=1, neg=0 to ensure crop with foreground and `Resize` to ensure your spatial size.
Hope it will helps you!

**Comment by maxencewynen** [comment]
Hey @KumoLiu, 
I think this could indeed solve my problem, thanks for your answer! 
I am wondering how I could handle the randomness of the zoom factor using this strategy, any idea? By that I mean that I always will have to specify the `spatial_size` parameter manually
A RandZoomByPosNegLabel would be really useful but I don't think it exists
Thanks again!

**Reply by KumoLiu** [reply]
Hi, I'm not really understand you, I think you only need to set the spatial size once in the instantiated. Like this
```
RandCropByPosNegLabeld(keys=["image", "label"], label_key="seg", spatial_size=(xx, xx), pos=1, neg=0),
Resized(keys=["image", "label"], spatial_size=(xx,xx))
```
Besides, RandZoomByPosNegLabel you suggest would perhaps need to consider how to crop and resize, so I guess that's enough for now. 
Thanks for your advice!

**Reply by maxencewynen** [reply]
Hi, I'm sorry I've not been clear enough.
My problem is specifically with the `spatial_size`  that I have to _specify_ . 
To better explain this, the `RandomZoomd `transform has 2 parameters called `min_zoom` and `max_zoom`, meaning the spatial size in that case is actually **randomly** taken between `image_size * min_zoom` and `image_size * max_zoom`. 
If I understood correctly, the `spatial_size ` parameter of `RandCropByPosNegLabeld` cannot be randomized, which I specifically need in order to perform a random zoom operation.
Hope this is clear now!

**Reply by KumoLiu** [reply]
Hi, I think your random zoom operation ia aim to keep original size. Am I right?
If like that, `RandCropByPosNegLabeld` can help you to random crop a patch and use the `Resize` with spatial_size as your original_size would help you keep your original size. Additional, you can sepcify the spatial_size in `RandCropByPosNegLabeld` with original_size/n which just mean your want to upsample your patches n times to your original shape.
